src/logging.c

```diff
@@ -414,6 +414,92 @@ static const char * copy_from_orig(
   return logging_output->buffer;
 }
 
+// copy buffers and decode the escape characters if they're exists
+static const char * decode_orig(
+  const logging_input_t * logging_input,
+  rcutils_char_array_t * logging_output,
+  size_t start_offset, size_t end_offset)
+{
+  size_t back_slash_index = rcutils_findn(
+    g_rcutils_logging_output_format_string + start_offset, '\\', end_offset - start_offset);
+  if (back_slash_index == SIZE_MAX) {
+    // back slash not found, copy full buffers
+    return copy_from_orig(logging_input, logging_output, start_offset, end_offset);
+  } else {
+    // TODO(anybody): optimize the following if possible
```

Comment:
this format initialization only happens once for the process space, so that would not affect the logging performance?

Reply:
Unfortunately, these handlers are called every time in `rcutils_logging_format_message` while logging messages.

Reply:
@fujitatomoya Thank you for bringing it to my attention. I'll update it to decode the escape characters only once.

Reply:
> that would not affect the logging performance?

Yes. addressed in https://github.com/ros2/rcutils/pull/426/commits/1bcdbda8c6b6a68960eb47d9a9990968eae74ed4

Reply:
@clalancette friendly ping

Reply:
moving this process in initialization makes sense to me. this decoding is different from token expanding so that it can be done only once during initialization.